pytest_httpserver/httpserver.py

```diff
@@ -675,6 +724,61 @@ def dispatch(self, request: Request) -> Response:
 
         return response
 
+    def _waiting_result_set(self, value: bool) -> None:
+        """Set waiting_result
+
+        Setting is implemented as putting value to queue without waiting. If queue is full we simply ignore the
+        exception, because that means that waiting_result was already set, but not read.
+        """
+        with suppress(queue.Full):
+            self._waiting_result.put_nowait(value)
+
+    def _waiting_result_update(self) -> None:
+        if not self.oneshot_handlers and not self.ordered_handlers:
+            self._waiting_result_set(True)
+
+    @contextmanager
+    def wait(self, raise_assertions: Optional[bool] = None, stop_on_nohandler: Optional[bool] = None,
+             timeout: Optional[float] = None):
+        """Context manager to wait until the first of following event occurs: all ordered and oneshot handlers were
+        executed, unexpected request was received (if `stop_on_nohandler` is set to `True`), or time was out
+
+        :param raise_assertions: whether raise assertions on unexpected request or timeout or not
+        :param stop_on_nohandler: whether stop on unexpected request or not
+        :param timeout: time (in seconds) until time is out
+
+        Example:
+            def test_wait(httpserver):
+                httpserver.expect_oneshot_request('/').respond_with_data('OK')
+                with httpserver.wait(raise_assertions=False, stop_on_nohandler=False, timeout=1) as waiting:
+                    requests.get(httpserver.url_for('/'))
+                # `waiting` is :py:class:`Waiting`
+                assert waiting.result
+                print('Processing time: {} - {} ({})'.format(waiting.start, waiting.stop, waiting.time))
+        """
+        self._waiting_settings.raise_assertions = (
+            self.default_waiting_settings.raise_assertions if raise_assertions is None else raise_assertions)
+        self._waiting_settings.stop_on_nohandler = (
+            self.default_waiting_settings.stop_on_nohandler if stop_on_nohandler is None else stop_on_nohandler)
+        self._waiting_settings.timeout = self.default_waiting_settings.timeout if timeout is None else timeout
+
+        # Ensure that waiting_result is empty
+        with suppress(queue.Empty):
+            self._waiting_result.get_nowait()
+
+        waiting = Waiting()
+        yield waiting
+
+        try:
+            waiting.complete(result=self._waiting_result.get(timeout=self._waiting_settings.timeout))
```

Comment:
Maybe moving `self._waiting_result.get()` into a separate line (between 772 and 773) would result a more readable code. I wanted to find where the `wait()` method will block and it was hard to find it for me.